test/cljs/chronograph/pages/create_organization/events_test.cljs

```diff
@@ -0,0 +1,128 @@
+(ns chronograph.pages.create-organization.events-test
+    (:require [cljs.test :refer-macros [deftest is testing run-tests]]
+              [day8.re-frame.test :as rf-test]
+              [re-frame.core :as rf]
+              [chronograph.specs]
+              [chronograph-web.pages.create-organization.events :as create-organization-events]
+              [chronograph-web.events.routing :as routing-events]
+              [chronograph-web.subscriptions :as subs]))
+
+(deftest create-organization-form-update-test
+  (testing "when the name is updated"
+    (rf-test/run-test-sync
+     (let [name "Nom"]
+       (rf/dispatch [::create-organization-events/create-organization-form-update :name name])
+       (is (= @(rf/subscribe [::subs/create-organization-form])
+              {:status :editing
+               :form-params {:name "Nom"}
+               :form-errors nil})))))
+
+  (testing "when the slug is updated"
+    (rf-test/run-test-sync
+     (let [slug "slug"]
+       (rf/dispatch [::create-organization-events/create-organization-form-update :slug slug])
+       (is (= @(rf/subscribe [::subs/create-organization-form])
+              {:status :editing
+               :form-params {:slug "slug"}
+               :form-errors nil})))))
+
+  (testing "when the name is invalid"
+    (rf-test/run-test-sync
+     (let [name ""]
+       (rf/dispatch [::create-organization-events/create-organization-form-update :name name])
+       (is (= @(rf/subscribe [::subs/create-organization-form])
+              {:status :editing
+               :form-errors {:name :invalid}
+               :form-params {:name ""}})))))
+
+  (testing "when the slug is invalid"
+    (rf-test/run-test-sync
+     (let [slug "SLU UG"]
+       (rf/dispatch [::create-organization-events/create-organization-form-update :slug slug])
+       (is (= @(rf/subscribe [::subs/create-organization-form])
+              {:status :editing
+               :form-errors {:slug :invalid}
+               :form-params {:slug "SLU UG"}}))))))
+
+(deftest create-organization-form-submit-test
+  (testing "when the organization is created successfully"
+    (rf-test/run-test-sync
+     (let [name "Name"
+           slug "slug"
+           id 83
+           organization {:id id :name name :slug slug}
+           token (atom nil)]
+       (rf/reg-fx :http-xhrio
+                  (fn [_]
+                    (rf/dispatch [::create-organization-events/create-organization-succeeded
+                                  organization])))
+       (rf/reg-fx :history-token #(reset! token %))
+       (rf/dispatch [::routing-events/set-page :organization-new])
```

Comment:
As mentioned earlier, we should only have to set the token. `pushy` should take care of calling `bidi` to set the page state.

Reply:
We've fixed this behaviour in  the normal path. Pushy wasn't routing when set-token was called because one can't call dispatch-sync from an event-handler.

However, we'll need access to window in our tests to do this. We've let that be a task for another day.